### PR TITLE
adding support for github auth for different tower instances

### PR DESCRIPTION
**Pull request by alanmoran**
cc @pmccarthy and @obrienrobert. I'll be raising a PR to the config repo to support this. Pushing here first so Eng/QE can look at applying the same changes to their private creds repo.

### CODE REVIEW

File: roles/credentials/tasks/template_github_credentials.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---  
+- set_fact: 
+    social_auth_github_org_key: "{{ lookup('vars', item + '_social_auth_github_org_key' ) }}"
+    social_auth_github_org_secret: "{{ lookup('vars', item + '_social_auth_github_org_secret' ) }}"
+    
+- include_tasks: encrypt_credentials.yml
```
Comment (pmccarthy) [inline roles/credentials/tasks/template_github_credentials.yml]: @alanmoran A task named `encrypt_credentials.yml` doesn't appear to exist?

Comment (alanmoran) [reply roles/credentials/tasks/template_github_credentials.yml]: @pmccarthy great catch. This must of changed name at some stage. In our repo it is still called encrypt_credentials.yml. It is now called encrypt_variables.yml.

**Comment by pmccarthy** [comment]
Ran the bootstrap process again from scratch and it completed successfully. One question though, I can see these variables still being generated:

```
social_auth_github_org_name
social_auth_github_org_key
social_auth_github_org_secret
social_auth_github_org_organization_map
social_auth_github_org_team_map
```

Is this to be expected? Can also see the `dummy_` equivalent variables too

**Comment by alanmoran** [comment]
> Ran the bootstrap process again from scratch and it completed successfully. One question though, I can see these variables still being generated:
> 
> ```
> social_auth_github_org_name
> social_auth_github_org_key
> social_auth_github_org_secret
> social_auth_github_org_organization_map
> social_auth_github_org_team_map
> ```
> 
> Is this to be expected? Can also see the `dummy_` equivalent variables too

No. I've fixed that now.

**Comment by pmccarthy** [comment]
@alanmoran Still seeing those original variables, see below:

```
grep -R social inventories/group_vars/
inventories/group_vars//all/tower_github_authentication.yml:social_auth_github_org_name: ''
inventories/group_vars//all/tower_github_authentication.yml:social_auth_github_org_key: ''
inventories/group_vars//all/tower_github_authentication.yml:social_auth_github_org_secret: ''
inventories/group_vars//all/tower_github_authentication.yml:social_auth_github_org_organization_map: ''
inventories/group_vars//all/tower_github_authentication.yml:social_auth_github_org_team_map: ''
inventories/group_vars//all/dummy_github_authentication.yml:dummy_social_auth_github_org_name: '<CHANGEME>'
inventories/group_vars//all/dummy_github_authentication.yml:dummy_social_auth_github_org_organization_map: "{'<CHANGEME>': {'admins': True, 'users': True}}"
inventories/group_vars//all/dummy_github_authentication.yml:dummy_social_auth_github_org_team_map: "{'<CHANGEME>': {'organization': u'<CHANGEME>', 'users': True}}"
inventories/group_vars//all/dummy_github_authentication.yml:dummy_social_auth_github_org_key: !vault
inventories/group_vars//all/dummy_github_authentication.yml:dummy_social_auth_github_org_secret: !vault
```

**Comment by alanmoran** [comment]
@pmccarthy Are you sure they are not from the last run? If you delete them and re-run does the file re-appear?

**Comment by pmccarthy** [comment]
@alanmoran Yep, I'm cleaning out the directory and checking out the branch each time. Just ran again to be sure and those vars are still in place.

Looks like they might be coming from `roles/credentials/tasks/template_github_credentials.yml`

**Review by StevenTobin — CHANGES_REQUESTED** [review]
That file is there because it exists in the base repo https://github.com/integr8ly/tower_dummy_credentials/blob/master/inventories/group_vars/all/tower_github_authentication.yml . Once this is rebased it lgtm

**Pull request merged**